### ISSUE TITLE
Support HDF5 reading/writing

### ISSUE THREAD

**Issue by sungho-cho** [hosseinmoein/DataFrame#326]
Input/output support for HDF5 (similar to pandas' [read_hdf](https://pandas.pydata.org/docs/reference/api/pandas.read_hdf.html) and [to_hdf](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_hdf.html)) would be immensely helpful to my team's project.

**Comment by sungho-cho** [comment]
Yeah sure, I know of these two open-source HDF5 libraries, although I just started trying them out as well.
- [h5cpp](https://github.com/ess-dmsc/h5cpp): C++ wrapper of the official HDF5 library
- [HighFive](https://github.com/BlueBrain/HighFive): Header only HDF5 interface

**Comment by hosseinmoein** [comment]
Implementing HDF5 from scratch on my side is not practical with the timeframe I have. Also I don't want to include a third part library as a dependency in _C++ DataFrame_.
But if you want to implement _C++ DataFrame_ read/write in HDF5 using one of the above libraries, I can show you how. It is not that involved.

**Comment by sungho-cho** [comment]
That's understandable. 
Looking at [load_data](https://htmlpreview.github.io/?https://github.com/hosseinmoein/DataFrame/blob/master/docs/HTML/load_data.html), I feel like it won't be too much of a hassle for me to write my own HDF5 reader and load it into a DataFrame.
I will reach out if there's any issues I need help with. Thanks!
